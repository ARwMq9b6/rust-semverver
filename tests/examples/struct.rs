#[allow(dead_code)]
pub mod old {
    pub struct Abc<A> {
        field: A,
    }

    pub struct Def {
        field: u8,
    }

    pub struct Efg {
        pub field: u8,
    }
}

#[allow(dead_code)]
pub mod new {
    pub struct Abc<B> {
        field: B,
    }

    pub struct Def<A=u8> {
        field: A,
    }

    pub struct Efg {
        pub field: u16,
    }
}
