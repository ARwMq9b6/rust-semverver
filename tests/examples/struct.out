old_ty: oldandnew::new::Def
new_ty: oldandnew::new::Def<A>
old_ty: oldandnew::new::Efg
new_ty: oldandnew::new::Efg
old_ty: oldandnew::new::Abc<B>
new_ty: oldandnew::new::Abc<B>
version bump: 1.0.0 -> (NonBreaking) -> 1.1.0
  TypeParameterAdded { defaulted: true }: Def
warning: change

